File: javascript/secret-handshake/secret-handshake.js
function SecretHandshake(input) {
	this.binaryInput = input.toString(2);
	this.result = [];
	if(this.binaryInput.length < 5){
		for (var i = 0;this.binaryInput.length<5; i++) {
			this.binaryInput = '0'+this.binaryInput;
		}
	}
	if (isNaN(this.binaryInput)) {throw new Error('Handshake must be a number');}
	this.commands = function () {
			for (var i = this.binaryInput.length - 1; i > 0; i--) {
			switch (i) {
				case 4: if (this.binaryInput[i] === '1') {this.result.push('wink');}
						break;
				case 3: if (this.binaryInput[i] === '1') {this.result.push('double blink');}
						break;
				case 2: if (this.binaryInput[i] === '1') {this.result.push('close your eyes');}
						break;
				case 1: if (this.binaryInput[i] === '1') {this.result.push('jump');}
						break;
				default: this.result.push('ERRO');
			}
			}
	if (this.binaryInput[0]==='1') {this.result = this.result.reverse();}
	return this.result;
	}
}

module.exports = SecretHandshake;

File: elixir/secret-handshake/secret_handshake.exs
defmodule SecretHandshake do
  @doc """
  Determine the actions of a secret handshake based on the binary
  representation of the given `code`.

  If the following bits are set, include the corresponding action in your list
  of commands, in order from lowest to highest.

  1 = wink
  10 = double blink
  100 = close your eyes
  1000 = jump

  10000 = Reverse the order of the operations in the secret handshake
  """
  #TODO make this in a less embaracing way
  require Bitwise
  @spec commands(code :: integer) :: list(String.t())
  def commands(code) do
    codes = [{0b1, "wink"}, {0b10, "double blink"}, {0b100, "close your eyes"}, {0b1000, "jump"}]
    l = Enum.reduce(codes, [], fn val, acc ->
      cond do
        Bitwise.band(code, elem(val, 0)) > 0 -> acc ++ [elem(val, 1)]
        true -> acc
      end
    end)
    if (code > 16) do
      Enum.reverse(l)
    else
      l
    end
  end
end
